tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -3065,4 +3065,85 @@ protected function assertEqualMarkup( $expected, $actual, $message = '' ) {
 	protected function add_html5_script_theme_support() {
 		add_theme_support( 'html5', array( 'script' ) );
 	}
+
+	/**
+	 * Test that a script is moved to the footer if it is made non deferrable, was in the header and
+	 * all scripts that depend on it are in the footer.
+	 *
+	 * @ticket 58599
+	 *
+	 * @dataProvider data_provider_script_move_to_footer
+	 *
+	 * @param string $b_in_footer in_footer value for script B.
+	 * @param string $expected    Expected output.
+	 * @param bool   $expect_a_in_footer Is script A in the footer?
+	 * @param bool   $expect_b_in_footer Is script B in the footer?
+	 */
+	public function test_wp_scripts_move_to_footer( $b_in_footer, $expected, $expect_a_in_footer, $expect_b_in_footer ) {
+		// Script A uses strategy=>defer. Script B depends on script A and uses in_footer=>true.
+		wp_enqueue_script( 'script-a', 'script-a.js', array(), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'script-b', 'script-b.js', array( 'script-a' ), null, $b_in_footer );
+
+		// Run the header and footer capturing the output.
+		ob_start();
+		wp_scripts()->do_head_items();
+		wp_scripts()->do_footer_items();
+		$actual = ob_get_clean();
+
+		$this->assertEquals( $expected, $actual );
+		$this->assertEquals( $expect_a_in_footer, in_array( 'script-a', wp_scripts()->in_footer, true ) );
+		$this->assertEquals( $expect_b_in_footer, in_array( 'script-b', wp_scripts()->in_footer, true ) );
```

Comment:
Instead of using `assertEquals`, you can now use the `assertEqualMarkup` method in this test case, which will ignore whitespace and unsemantic differences (e.g. attribute quote style).

Reply:
I tried changing this and had issues, leaving as is since tests pass

Reply:
feel free to update!